### PR TITLE
keep focus on multi-select when using tab

### PR DESCRIPTION
**Pull request by bboyle**
For consideration. A possible start to dealing with some of the requests for different `Tab` behaviour on multi-select controls. Limited to multi-select and auto-close=false to avoid creating a keyboard trap. Feedback welcome.

in a multi-select control with auto-close = false:

- Tab functions like Enter (selects item, keeps focus on select control)
- Closing the options (Escape) allows the default Tab behaviour (change
  focus)

### CODE REVIEW

**Comment by jgentes** [comment]
Be sure to address the issue identified here before merging this to the main branch: https://github.com/angular-ui/ui-select/issues/1261#issuecomment-160681802

**Comment by bboyle** [comment]
@jgentes will do

**Comment by synthecypher** [comment]
This getting merged anytime soon?

**Comment by bboyle** [comment]
@synthecypher priority dropped on this where I work so I haven't got back to it. If you've time to work on the feedback from @jgentes that would be a great help!

**Comment by aaronroberson** [comment]
Any progress @synthecypher or @bboyle?

File: test/select.spec.js [hunk 1]
```diff
@@ -1799,6 +1811,56 @@ describe('ui-select tests', function() {
...
+        spyOn(jQuery.Event.prototype, 'stopPropagation');
+
+        triggerKeydown(searchInput, Key.Tab);
+        expect(jQuery.Event.prototype.preventDefault).not.toHaveBeenCalled();
+        expect(jQuery.Event.prototype.stopPropagation).not.toHaveBeenCalled();
+
+    });
+
+    it('should stop the propagation when pressing TAB key from dropdown when control is open', function() {
+        window.console.log( 'TAB while multi-select open' );
```
Comment (wesleycho) [inline test/select.spec.js]: Remove console.log

Comment (bboyle) [inline test/select.spec.js]: done

**Comment by wesleycho** [comment]
History needs to be fixed - there should not be any merge commits.

**Comment by bboyle** [comment]
@wesleycho I do not understand the "history needs to be fixed" comment. I merged the latest master so this branch was not sooo far behind, was that bad?

**Comment by wesleycho** [comment]
Proper approach would have been to rebase - merge commits mutate order of history in undesirable ways, which has a strong negative effect in open source.

**Comment by bboyle** [comment]
hmmm, sorry but that is over my head in git knowledge. perhaps somebody can fix it, or might need to reject this PR.

**Comment by wesleycho** [comment]
In this case, you might be able to salvage it in this branch by doing `git reset --hard 5fdd1e5; git pull --rebase origin master`, resolve the conflicts, then do `git add -A; git rebase --continue`, redo the removal of the console.log, then `git push -f`.

**Comment by bboyle** [comment]
cool, thanks for the tips... no conflicts so that's done with the force push

**Comment by wesleycho** [comment]
console.log needs to be removed again - once you remove it, you can commit via `git add -A; git commit --amend; git push -f`

**Comment by wesleycho** [comment]
Hmm, there is a merge commit again.

If this is to be merged, this may have to be manually cherry-picked.

**Comment by bboyle** [comment]
@wesleycho is a new PR ok? would that be easier than cleaning the git history here?

**Comment by wesleycho** [comment]
Yeah, I think that might be best in this situation.

**Comment by aaronroberson** [comment]
@bboyle thanks for being such a good sport!

**Comment by bboyle** [comment]
@wesleycho clean PR is here https://github.com/angular-ui/ui-select/pull/1534

@aaronroberson it is a bit complex, one day I will have to learn the right ways to use git pull and git rebase

**Pull request closed without merging**